### PR TITLE
migrated two tests from ami-val project

### PR DESCRIPTION
**Pull request by jabia99**
Migrated two basic tests from ami-val

### CODE REVIEW

File: test_suite/generic/test_generic.py [hunk 1]
```diff
@@ -135,6 +135,14 @@ class TestsNetworking:
     def test_dns_resolving_works(self, host):
         host.run_test('ping -c 5 google-public-dns-a.google.com')
 
+    def test_ipv_localhost(self, host):
+        expected_hosts = ['127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4',
+                          '::1         localhost localhost.localdomain localhost6 localhost6.localdomain6']
+        with host.sudo():
+            for expected_host in expected_hosts:
+                assert host.file('/proc/cmdline').contains(expected_host), \
+                    "/etc/hosts does not contain ipv4 or ipv6 localhost"
```
Comment (narmaku) [inline test_suite/generic/test_generic.py:144]:
```suggestion
                    '/etc/hosts does not contain ipv4 or ipv6 localhost'
```

File: test_suite/generic/test_generic.py [hunk 2]
```diff
@@ -135,6 +135,14 @@ class TestsNetworking:
     def test_dns_resolving_works(self, host):
         host.run_test('ping -c 5 google-public-dns-a.google.com')
 
+    def test_ipv_localhost(self, host):
+        expected_hosts = ['127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4',
+                          '::1         localhost localhost.localdomain localhost6 localhost6.localdomain6']
+        with host.sudo():
+            for expected_host in expected_hosts:
+                assert host.file('/etc/hosts').contains(expected_host), \
```
Comment (narmaku) [inline test_suite/generic/test_generic.py:143]: I think we are reading the wrong file here :sweat_smile:

**Review by narmaku — CHANGES_REQUESTED** [review]
Nice. Some minor comments!

File: test_suite/generic/test_generic.py (hunk 2 above)
Comment (jabia99) [reply test_suite/generic/test_generic.py:143]: why?

File: test_suite/cloud/test_aws.py [hunk 3]
```diff
@@ -201,6 +201,16 @@ def test_max_cstate_is_configured_in_cmdline(self, host):
...
                         f'{line} must not be specified in AMIs that are not SAP'
 
+    def test_hostkey_permissions(self, host):
+        """
+        BugZilla 2013644
+        """
+        files_to_check = ['ssh_host_ecdsa_key', 'ssh_host_ed25519_key', 'ssh_host_rsa_key']
+        for file in files_to_check:
+            if host.file(f'/etc/ssh/{file}').exists:
+                assert host.file('/etc/ssh/').mode >= 0o640, \
```
[the remark this answers is not in the document]
Comment (narmaku) [reply test_suite/cloud/test_aws.py:211]: Question: Does the `0o640` mode also check the group and the owner (I think root/root was the unwanted one)?

If not, we will have to add that check too (see File.user and File.group).

**Review by narmaku — CHANGES_REQUESTED** [review]

File: test_suite/cloud/test_aws.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jabia99) [reply test_suite/cloud/test_aws.py:211]: oh right, I think it does get checked but it doesn't make sense to do it that way

**Review by narmaku — APPROVED** [review]
LGTM!

**Pull request merged**